File: src/main/java/myLang/interpreter/Function.java
package myLang.interpreter;

import myLangParser.MyLangParser;

import java.util.LinkedList;

/**
 * Класс обертка, хранит информацию о функции:
 * <ol>
 *     <li>имя {@code name}</li>
 *     <li>возвращаемое значение {@code resultType}</li>
 *     <li>список аргументов {@code args}</li>
 *     <li>ссылка на блок метода {@code ctx}</li>
 * </ol>
 */
public class Function {
    final String name, resultType;
    final LinkedList<FunArgs> args;
    final MyLangParser.FunDeclarationContext ctx;

    Function(String name, String resultType, LinkedList<FunArgs> args, MyLangParser.FunDeclarationContext ctx) {
        this.name = name;
        this.resultType = resultType;
        this.args = args;
        this.ctx = ctx;
    }

    public String getName() {
        return name;
    }

    public String getResultType() {
        return resultType;
    }

    public LinkedList<FunArgs> getArgs() {
        return args;
    }

    public MyLangParser.FunDeclarationContext getCtx() {
        return ctx;
    }

}

File: src/main/kotlin/myLangKotlin/interpreter/Function.kt
package myLangKotlin.interpreter

import myLangParser.MyLangParser

/**
 * Класс обертка, хранит информацию о функции:
 *
 *  1. имя `name`
 *  2. возвращаемое значение `resultType`
 *  3. список аргументов `args`
 *  4. ссылка на блок метода `ctx`
 *
 */
class Function internal constructor(val name: String, val resultType: String, val args: List<FunArgs>, val ctx: MyLangParser.FunDeclarationContext)
